srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/ProcedureCsvReader.java

```diff
@@ -0,0 +1,278 @@
+     * returns an empty string for a missing value.
+     * @return the value at the specified index, if it exists, or an empty string
+     * otherwise
+     */
+    private String tryGetValue(final CSVRecord record, final int i)
+    {
+        return (record.size() > i) ? record.get(i) : "";
+    }
+    
+    /**
+     * Parses a Y/N string into a boolean value.
+     * @param s must be "Y" or "N" (case-insensitive)
+     * @return true for "Y", false for "N"
+     * @throws IllegalArgumentException if s is not "Y" or "N"
+     */
+    private boolean parseCsvBoolean(final String s)
+    {
+        if (s.equalsIgnoreCase("Y"))
+        {
+            return true;
+        }
+        else if (s.equalsIgnoreCase("N"))
+        {
+            return false;
+        }
+        else
+        {
+            throw new IllegalArgumentException("The String value must be Y or N.");
+        }
+    }
+    
+    /**
+     * Translates a single CSVRecord into a single Procedure object.
+     * @param record the CSVRecord to translate
+     * @param errors to record validation errors
+     * @param rowNumber the row number for any validation errors (1-based)
+     * @return the translated Procedure, if there were no validation errors
+     */
+    private Optional<Procedure> translateRecord(
+            final CSVRecord record,
+            final Collection<TabularUploadError> errors,
+            final int rowNumber)
+    {
+        // Construct our own error list to detect if we added any errors.
+        final ArrayList<TabularUploadError> ourErrors = new ArrayList<>();
+
+        /* Get all the string values first since they don't involve exceptions */
+        
+        final String cptCode = tryGetValue(record, 0);
+        if (cptCode.length() != Procedure.CPT_CODE_LENGTH)
+        {
+            ourErrors.add(TabularUploadError.forField(
+                    rowNumber,
+                    "cptCode",
+                    String.class,
+                    ValidationCodes.BAD_FIXED_LENGTH,
+                    new Object[] { Procedure.CPT_CODE_LENGTH },
+                    "invalid length"));
+        }
+        
+        final String complexity = tryGetValue(record, 3);
+        ValidationUtils2.validateRequiredString(
+                complexity, rowNumber, "complexity", Procedure.COMPLEXITY_MAX, ourErrors);
+        
+        final String longDesc = tryGetValue(record, 4);
+        ValidationUtils2.validateRequiredString(
+                longDesc, rowNumber, "longDescription", Procedure.DESCRIPTION_MAX, ourErrors);
+        
+        final String shortDesc = tryGetValue(record, 5);
+        ValidationUtils2.validateRequiredString(
+                shortDesc, rowNumber, "shortDescription", Procedure.DESCRIPTION_MAX, ourErrors);
+        
+        final String rvuString = tryGetValue(record, 1);
+        float rvu = Float.NaN;
+        try
+        {
+            rvu = Float.parseFloat(rvuString);
```

Comment:
There is a Floats.tryParse() method from Guava that doesn't require an exception to be caught.